app/services/notification_service.rb

```diff
@@ -0,0 +1,43 @@
+class NotificationService
+  def initialize(conference_room)
+    @client = GoogleOauth.push_notification_client
+    @conference_room = conference_room
+    @channel = conference_room.channel
+  end
+
+  def renew_subscription
+    return create_subscription if channel.nil?
+    unsubscribe
+    subscribe
+  end
+
+  def unsubscribe
+    client.stop_channel(google_channel)
+  end
+
+  private
+
+  attr_accessor :conference_room, :channel, :client
+
+  def create_subscription
+    create_new_channel
+    subscribe
+  end
+
+  def subscribe
+    new_google_channel = google_subscription_channel
+    channel.google_update!(new_google_channel)
+  end
+
+  def create_new_channel
```

Comment:
I'd change the method name to `build_new_channel` as you don't save it in the DB here